### PR TITLE
Feature: Tags

### PR DESCRIPTION
**Pull request by zakariaelas**
See #174 for more information about the initial discussion. 

@yjose one thing to note: Initially, we had a ton of tags. This is because of a couple of reasons:

1. Because of how the `group` works in GraphQL queries in Gatsby, it is case sensitive. This means that `Dev` and `dev` result in two different tags.
2. The tags have not been consistent across episodes. Tags were written in different ways when they mean the same thing. (ex: mobile and mobile dev, react native and ReactNative, etc.)
3. Sometimes, tags were a bit too specific. Ex: `women in tech`. I think we can just use something short and clear like `women`, and it would represent the same thing.

I went through all the episodes and tried my best to fit most episodes under a general set of tags. I believe it became somewhat acceptable, but feel free to narrow it down further. 

If you approve the feature in general, I suggest we try to follow a standard in future episodes. I can add a section in the docs about this. For example: 

> Use all uppercase letters for acronyms like AI, ML, MSS, etc. 
> Use all lowercase letters for abbreviations and words like dev, frontend, react native, etc.

### CODE REVIEW

**Comment by yjose** [comment]
@ismailElazizi would love to hear your thoughts about tags, we have a lot of tags, do you think this is the best way to do  it https://deploy-preview-182--geeksblabla.netlify.app/blablas

**Comment by soufianelf** [comment]
Can we hide the tags and let the user display them when he need them

**Comment by yjose** [comment]
Good approach @soufianelf, i would suggest to show first tags  line  and ••• To show and hide others. What you think @zakariaelas

**Comment by zakariaelas** [comment]
Good idea @soufianelf @yjose , a toggle would be a good solution. Small concern from a technical side, we will need to find a way to persist the toggle state between the different pages (`/blablas`, `/${tag}`/, etc.). I don't have a lot of experience with gatsby, but I believe we will need to use a gatsby browser API (wrapPageElement). Is that okay ?

**Comment by yjose** [comment]
yeah you can try to add it a theme state, theme component is already in the root  https://github.com/DevC-Casa/geeksblabla.com/blob/master/gatsby-browser.js

**Comment by zakariaelas** [comment]
Great, I'll work on it as soon as I have some time this week.

**Comment by yjose** [comment]
Guys, just come in my mind to use categories instead of tags as for some tags we only have 1 or 2 episode and the page look wired. 

I would suggest to use categories instead, in such case we will have 4 categories : career, dev, mss, AMA 

@soufianelf @zakariaelas what you think?

**Comment by zakariaelas** [comment]
@yjose i think it's better than using tags, because there are so many with only 1 or 2 episodes like you said. 

Some questions so I can understand your idea better:

1. Should we add a category field in the episode frontmatter? Personally I think it's better to do it this way so it becomes easy to add a category in the future.
2. Should we create a page for each category i.e: `/career/`? I think we should so that people can share relevant categories with others or even bookmark them.
3. What will happen to the tags field in the frontmatter of episodes? Should we leave them? 

What do you guys think?

**Comment by yjose** [comment]
1. Yes 
2. Yes
2. I think we should leave them, as we are using them to generate cover add keywords SEO

**Review by iMeriem — APPROVED** [review]

**Comment by zakariaelas** [comment]
@yjose finally got some time to work on your suggestion, please take a look and let me know what you think :pray:

File: blablas/ep7/index.md [hunk 1]
```diff
@@ -3,7 +3,8 @@ date: 2018-10-25
 time: 20h
 duration: "1:22:38"
 title: "Métier de développeur"
-tags: ["dev"]
+tags: ["dev", "career"]
+category: "MSS"
```
Comment (yjose) [inline blablas/ep7/index.md:7]: should be career

**Comment by yjose** [comment]
LGTM, Thank you @zakariaelas for the great work

**Review by soufianelf — APPROVED** [review]

**Comment by soufianelf** [comment]
Sounds good we can think more about the categories later we can add more categories like software security and entrepreneurship

File: blablas/ep7/index.md (hunk 1 above)
Comment (zakariaelas) [reply blablas/ep7/index.md]: @yjose my bad :laughing: . Fixed now :+1:

**Comment by zakariaelas** [comment]
@soufianelf sounds good. I have also updated the docs to let contributors know about the category field. Also, I have provided a standard way for everyone to add tags to episodes, just in case you would want to use tags one way or another in the future. It's better to stay consistent across all episodes.

**Comment by soufianelf** [comment]
awesome thanks @zakariaelas

**Review by yjose — APPROVED** [review]
👍

**Comment by yjose** [comment]
@zakariaelas  you twitter handler or Facebook profile, please

**Comment by zakariaelas** [comment]
@yjose 
@21kelas on twitter, Zakaria El Asri on facebook

**Pull request merged**